### ISSUE TITLE
Add ways to load GLTF files in the browser version of Filament

### ISSUE THREAD

**Issue by Antonio-Sorrentini** [google/filament#1843]
After some days of struggling I have to ask if it is possible to enhance the allowed ways to load GLTF files in the webgl version of filament. 

At the moment one can only load gltf files from a remote server and almost only from the root path, unless one embeds everything in a single file. 

It's not possibile to load gltf files from the local file system neither using input or drag and drop in the browser. I've tried everything, translating images and bin files in blobs and changing the gltf file on the spot putting inside them createObjectURL, but it does not work because Filament doesn't like that kind of urls despite them working very well in the browser. I also tried converting every resource in a data url, this seems to land to some sort of success but still very limited because for reasons I cannot understand if images are very big things get weird, and sometimes in some browsers they load in others not and so on. 

I'm trying to write a sort of 3D modeler for the web using Filament WebGL version. Having users to be able to simply drag and drop GLTF files in the browser (like you do with all the other GLTF viewers from threejs, babylonjs and so on) is a must. 

I've also heavily tried to hack the internal fetch function and resourcesLoader javascript stuff, but with only limited success, and after all those would only be hacks, no good. And anyway it is very hard because most of the code is wasm, at that point maybe it's better to move to the C++ sources. 

I see Filament has still a lot of these limitations, I'm talking front webgl. I can probably try to contribute to enhance these aspects if there is an overall interest from the project leaders in having Filament to be first class citizen also on the web.

**Comment by prideout** [comment]
You can already pass a JavaScript [ArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) directly into `createAssetFromBinary` rather than a URL.  You can also write your own version of our [loadResources](https://github.com/google/filament/blob/0e81207758d82ed75b44674fa5a63be550c631ee/web/filament-js/extensions.js#L258) utility, which is written in JavaScript rather than C++.  This means it is theoretically possible to write a drag-and-drop viewer without any changes to the Filament tree.

If I have time I might be able to write a sample app for this.

**Comment by Antonio-Sorrentini** [comment]
Thanks @prideout. I have no problem in loading .glb or .gltf files directly on my own. I simply encode their content (which was loaded on my own from user file or even created on the spot) in a Uint8Array buffer and then I exec Filament.assets[uri]=<that-Uint8Array-buffer> then I call engine.createAssetLoader(); and loader.createAssetFromBinary(uri)  or  loader.createAssetFromJson(uri);
and then asset.loadResources(finalizer -> ....);

This works well, but only if all the the resources are embeded inside the encoded file. If for example the gltf file contains uri for images and bin files, etc. this does not work. I tried to put them (these other resources) also in Filament.assets[uri]=<Uint8Array-buffer-of-images-and-the-like> before calling createAssetLoader and the rest, but it does not work. As for now the best way of doing this has been demostrated for me to encode everything in data urls and modify on the spot the original gltf file to changed the actual uri with the dara url made by myself. This works. 

You say that I can use createAssetFromBinary but I cannot understand how. I need to have createAssetFrom<..> to find images and bin files without Filament fetching them on its own from uris, and instead using my own provided Uint8Array which I already populated using user files from drag and drop or input field, and possibily without having me to modify the original files and replace the uri with dara uri.
